salt/modules/dummyproxy_package.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+'''
+Package support for the dummy proxy used by the test suite
+'''
+from __future__ import absolute_import
+
+# Import python libs
+import logging
+import salt.utils
+
+
+log = logging.getLogger(__name__)
+
+# Define the module's virtual name
+__virtualname__ = 'pkg'
+
+
+def __virtual__():
+    '''
+    Only work on systems that are a proxy minion
+    '''
+    try:
+        if salt.utils.is_proxy() and __opts__['proxy']['proxytype'] == 'dummy':
+            return __virtualname__
+    except KeyError:
+        return (False, 'The dummyproxy_package execution module failed to load.  Check the proxy key in pillar or /etc/salt/proxy.')
```

Comment:
TBH, I'm not sure how to resolve this error message as a potential new user. Could it be improved?